cli/flox/src/commands/activate.rs

```diff
@@ -375,118 +399,48 @@ impl Activate {
-    /// to put the system curated path items first in the `PATH`,
-    /// which completely breaks the user's ability to manage their `PATH` in subshells,
-    /// e.g. when using tmux.
-    ///
-    /// On macos `/usr/libexec/path_helper` is typically invoked from
-    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
-    ///
-    /// Note: since the "path_helper" i only invoked by login shells,
-    /// we only need to setup the PATH patching for `flox activate` in shell rc files.
-    ///
-    /// ## Example
-    ///
-    /// > User has `eval "$(flox activate)"` in their `.zshrc`.
-    ///
-    ///  Without the path patching, the following happens:
-    ///
-    /// 1. Open a new terminal (login shell)
-    ///     -> `path_helper` runs (`PATH=<default envs>`)
-    ///     -> `flox activate` runs (`PATH=<flox_env>:<default envs>`)
-    /// 2. Open a new tmux session (login shell by default)
-    ///     -> `path_helper` runs (`PATH=<default envs>:<flox_env>`)
-    ///     -> `flox activate` runs
-    ///     -> ⚡️ environment already active, activate skipped
-    ///        without path patching: `PATH:<default envs>:<flox_env>` ❌
-    ///        with path patching: `PATH:<flox_env>:<default envs>`    ✅ flox env is not shadowed
-    ///
-    /// ## Implementation
-    ///
-    /// This function attempts to undo the effects of `/usr/libexec/path_helper`
-    /// by partitioning the `PATH` into two parts:
-    /// 1. known paths of activated flox environments
-    ///    and nix store paths (e.g. `/nix/store/...`) -- assumed to be `nix develop` paths
-    /// 2. everything else
+    /// Activate the environment interactively by spawning a new shell
+    /// and running the respective activation scripts.
     ///
-    /// The `PATH` is then reordered to put the flox environment and nix store paths first.
-    /// The order within the two partitions is preserved.
-    #[cfg_attr(not(target_os = "macos"), allow(unused_variables))] // on linux `flox_env_dirs` is not used
-    fn fixup_path(flox_env_dirs: &IndexSet<PathBuf>) -> Option<Result<OsString>> {
-        #[cfg(not(target_os = "macos"))]
-        {
-            None
-        }
-        #[cfg(target_os = "macos")]
-        {
-            if !Path::new("/usr/libexec/path_helper").exists() {
-                return None;
-            }
-            let path_var = env::var("PATH").unwrap_or_default();
-            let fixed_up_path = Self::fixup_path_with(path_var, flox_env_dirs);
-            let fixed_up_path_joined = env::join_paths(fixed_up_path).context(
-                "Cannot activate environment because its path contains an invalid character",
-            );
-            Some(fixed_up_path_joined)
+    /// This function should never return as it replaces the current process
+    fn activate_interactive(
+        shell: Shell,
+        exports: HashMap<&str, String>,
+        activation_path: PathBuf,
+        now_active: UninitializedEnvironment,
```

Comment:
I don't love this combination of argument name and type. What does it mean for `now_active` to be an `UninitializedEnvironment`? I know this is probably an existing convention in the codebase, but if we're doing a rewrite we may as well use good names.

Reply:
My approach with this effort was to remain entirely consistent with what was there before, primarily so that we can all more readily recognise what has and hasn't been changed. In this case I copied the function signature and variable names from the previous version - until we drop support for activating old rendered environments (and remove the `old_activate_*()` functions accordingly) I'd be inclined to keep that part of things the same. Make sense?

Reply:
FWIW `UninitializeEnvironment` is basically just a descriptor of an environment. "Uninitialized" because it represents an environment but does not attempt to load any of the related files.
I think we had some previous discussion on how to name this "Unverified", or "EnvironmentDescriptor" come to mind but I agree that this probably shouldnt be included in this PR